src/libmongoc/src/mongoc/mongoc-cyrus.c

```diff
@@ -31,6 +31,11 @@
 #undef MONGOC_LOG_DOMAIN
 #define MONGOC_LOG_DOMAIN "CYRUS-SASL"
 
+// CDRIVER-2722: the entire Secure Transport library is deprecated.
+#if defined(MONGOC_ENABLE_SSL_SECURE_TRANSPORT)
```

Comment:
Missed this earlier, but the Cyrus warnings were unrelated to Secure Transport.

Reply:
Added proper distinction between Cyrus SASL (cross-platform, but specifically for MacOS) and the Secure Transport library (MacOS only). Verified by [this patch](https://spruce.mongodb.com/version/67a6674c7e6ae50007bd47b2/).